src/lib/pubkey/ec_group/ec_inner_data.h

```diff
@@ -259,23 +268,49 @@ class EC_Group_Data final : public std::enable_shared_from_this<EC_Group_Data> {
          return *m_pcurve;
       }
 
+      /**
+      * Note this constructor should *only* be called by EC_Group_Data::create.
+      *
+      * It is only public to allow use of std::make_shared
+      */
+      EC_Group_Data(const BigInt& p,
+                    const BigInt& a,
+                    const BigInt& b,
+                    const BigInt& g_x,
+                    const BigInt& g_y,
+                    const BigInt& order,
+                    const BigInt& cofactor,
+                    const OID& oid,
+                    EC_Group_Source source);
```

Comment:
I would tend to agree on using `new` rather than exposing a constructor we don't want called, but in this case the type is entirely internal, so a comment seemed sufficient.